### DISCUSSION TITLE
Using Cloudflare warp

### DISCUSSION THREAD

**Discussion by EashanC23** [FelixKratz/SketchyBar#325] [Q&A]
I'm trying to integrate cloudflare warp + cloudflare-warp cli with my sketchybar that uses an alias and has a simple click script that toggles it on click. However, with the given code block below, it only turns it on and will not turn off, when sketchybar is running as a service. If it runs as a regular command, the toggling works fine, the problem is when it is a homebrew service. 

sketchybarrc
```
sketchybar --add alias "Cloudflare WARP" right \
           --rename "Cloudflare WARP" warpAlias \
           --set warpAlias click_script="$PLUGIN_DIR/warp.sh"\
                            icon.padding_right=0\
                            background.padding_right=-4\
                            label.padding_right=0
```

warp.sh
```
#!/bin/bash


status=$(warp-cli status | grep nected)
status=${RAA#"Status update: "}
if [[ $status != "Connected" ]];then
	# echo "here"
	/usr/local/bin/warp-cli connect
else
	/usr/local/bin/warp-cli disconnect 
fi
``` 

Any help is appreciated !

**Comment by FelixKratz** [comment]
What I am wondering is what is the content of `RAA` and where is it set? Because it seems to me that the `status` variable is overwritten by an emtpy string which always falls through the loop condition if `RAA` is not set, i.e. what is the purpose of the first two lines?

**Reply by EashanC23** [reply]
Oh, very sorry the ```RAA``` variable should be the ```status``` variable, I just forgot to rename it, but even then it still was not working. I had just renamed it because I was embarassed of my variable naming. 😅 Assume the ```RAA``` variable to be the ```status``` variable 

warp.sh
```
status=$(warp-cli status | grep nected)
status=${status#"Status update: "}
if [[ $status != "Connected" ]];then
	# echo "here"
	/usr/local/bin/warp-cli connect
else
	/usr/local/bin/warp-cli disconnect 
fi
```

**Reply by FelixKratz** [reply]
Can you try running this from the your commandline:
```bash
env -i HOME="$HOME" sh -c "$HOME/.config/sketchybar/plugins/wake.sh" 
```
Does it toggle the status correctly? This basically creates a clean environment, sets the home dir and runs the script in a clean shell. This is basically what sketchybar does when run as a service.

**Reply by FelixKratz** [reply]
Probably not, because `warp-cli` is not in `PATH` in the first line, try replacing it in the first line with: `/usr/local/bin/warp-cli`

**Reply by EashanC23** [reply]
Ah yes replacing it with ```/usr/local/bin/warp-cli``` worked! Help was very much appreciated as I'm still new to bash scripting.
